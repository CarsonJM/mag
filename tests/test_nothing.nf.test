nextflow_pipeline {

    name "Test pipeline"
    script "../main.nf"
    profile "test_nothing"

    test("Profile: test_nothing") {

        when {
            params {
                outdir = "$outputDir"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    file("$outputDir/multiqc/multiqc_data/").list().sort(),
                    file("$outputDir/multiqc/multiqc_plots/").list().sort(),
                    file("$outputDir/QC_shortreads/fastqc/").list().sort(),
                    path("$outputDir/QC_shortreads/remove_phix/")
                ).match() },
                { assert new File("$outputDir/multiqc/multiqc_report.html").exists() },
                { assert new File("$outputDir/pipeline_info/nf_core_pipeline_software_mqc_versions.yml").exists() }
            )
        }
    }
}
