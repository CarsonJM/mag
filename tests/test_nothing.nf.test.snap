{
    "Profile: test_nothing": {
        "content": [
            [
                "multiqc.log",
                "multiqc_bowtie2.yaml",
                "multiqc_citations.yaml",
                "multiqc_data.json",
                "multiqc_fastqc.yaml",
                "multiqc_fastqc_1.yaml",
                "multiqc_general_stats.yaml",
                "multiqc_software_versions.yaml",
                "multiqc_sources.yaml"
            ],
            [

            ],
            [
                "test_minigut_run0_raw_1_fastqc.html",
                "test_minigut_run0_raw_2_fastqc.html",
                "test_minigut_run0_trimmed_1_fastqc.html",
                "test_minigut_run0_trimmed_2_fastqc.html",
                "test_minigut_sample2_run0_raw_1_fastqc.html",
                "test_minigut_sample2_run0_raw_2_fastqc.html",
                "test_minigut_sample2_run0_trimmed_1_fastqc.html",
                "test_minigut_sample2_run0_trimmed_2_fastqc.html"
            ],
            [
                "test_minigut_run0_phix_removed.bowtie2.log:md5,4f10d4bc2b3f0be722807bbf16659621",
                "test_minigut_sample2_run0_phix_removed.bowtie2.log:md5,f1530d72fd5626bb98c60400b294dc18"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.1"
        },
        "timestamp": "2024-05-23T14:59:09.216485918"
    }
}
